p2p/exchange.go

```diff
@@ -157,8 +167,15 @@ func (ex *Exchange[H]) Head(ctx context.Context, opts ...header.HeadOption[H]) (
 	)
 	for _, from := range peers {
 		go func(from peer.ID) {
+			_, newSpan := span.TracerProvider().Tracer("requesting peer").Start(
+				ctx, "",
+				trace.WithAttributes(attribute.String("peerID", from.String())),
+			)
+			defer newSpan.End()
+
 			headers, err := ex.request(reqCtx, from, headerReq)
```

Comment:
optional nit: it's correct to pass the ctx with span there so that `request` can see it and be a child if we ever add span there inside.

Reply:
also, the span would take the reqCtx.

Reply:
also, why not to use tracerClient directly?

Reply:
do you mean original span?